### PR TITLE
Wind weather overhaul

### PR DESCRIPTION
**Pull request by ulteq**
* Added real world weather data for Stratis/Altis (https://weatherspark.com/averages/32194/Lemnos-Limnos-North-Aegean-Islands-Greece)
* Added temperature and humidity randomization on mission start
* Overworked the client side getWind function for more dynamic wind speeds
* Minimum and maximum wind speeds are calculated on mission start
* Reference wind direction is calculated on mission start

### CODE REVIEW

**Comment by jaynus** [comment]
Are these changes randomizing wind again between two clients with those getWind changes? It looks like they are. This completely negates the whole purpose of this; which was to make sure wind speeds are synchronized between a spotter and sniper. 

Are the variances small enough between clients?

**Comment by esteldunedain** [comment]
No, the randomization is done on the server and broadcasted, similar to the old system. It's just that client side computation is more rich (more variations), while still being synced.

**Comment by jaynus** [comment]
Then disregard my comment, I'm good with it.

**Comment by sunrrrise** [comment]
Is it correct/by design that changing weather by Zeus or MCC is no longer possible?

**Comment by jaynus** [comment]
> Is it correct/by design that changing weather by Zeus or MCC is no longer possible? 

Correct. We have not added the appropriate calls in those associated modules yet.

**Comment by ulteq** [comment]
> Is it correct/by design that changing weather by Zeus or MCC is no longer possible?

This pull request is focused on improving the wind based on real world data. Data for Stratis/Altis:
ACE_WindSpeedMax[]  = {{8.8, 5.5}, {8.8, 5}, {8.6, 4.8}, {7.6, 3.4}, {7.0, 3.0}, {7.1, 3.0}, {7.5, 3.1}, {8.0, 3.2}, {7.6, 3.5}, {7.8, 4.6}, {7.9, 5.0}, {8.2, 5.5}};
ACE_WindSpeedMean[] = {4.8, 4.9, 4.6, 4.1, 3.5, 3.5, 4.3, 4.4, 4.1, 4.5, 4.5, 5.0};
ACE_WindSpeedMin[]  = {{0.2, 5.0}, {0.1, 5.0}, {0.2, 4.3}, {0.0, 3.0}, {0.0, 2.1}, {0.0, 2.0}, {0.1, 3.1}, {0.3, 3.1}, {0.0, 3.6}, {0.0, 4.2}, {0.1, 5.0}, {0.2, 5.5}};
ACE_WindDirectionProbabilities[]  = {{0.06, 0.32, 0.05, 0.04, 0.15, 0.06, 0.02, 0.02}, // January
                                     {0.08, 0.32, 0.04, 0.04, 0.18, 0.06, 0.02, 0.02}, // February
                                     {0.09, 0.30, 0.04, 0.04, 0.20, 0.06, 0.02, 0.03}, // March
                                     {0.10, 0.25, 0.03, 0.04, 0.22, 0.06, 0.02, 0.04}, // April
                                     {0.18, 0.25, 0.03, 0.04, 0.18, 0.04, 0.01, 0.05}, // May
                                     {0.25, 0.25, 0.03, 0.03, 0.15, 0.03, 0.00, 0.08}, // June
                                     {0.32, 0.30, 0.02, 0.02, 0.10, 0.01, 0.00, 0.09}, // July
                                     {0.28, 0.35, 0.02, 0.01, 0.08, 0.01, 0.00, 0.08}, // August
                                     {0.20, 0.37, 0.03, 0.01, 0.11, 0.01, 0.01, 0.05}, // September
                                     {0.10, 0.39, 0.04, 0.02, 0.15, 0.02, 0.01, 0.03}, // October
                                     {0.08, 0.38, 0.06, 0.04, 0.19, 0.03, 0.02, 0.02}, // November
                                     {0.06, 0.37, 0.05, 0.03, 0.18, 0.04, 0.02, 0.02}};// December

File: addons/weather/XEH_postInit.sqf [hunk 1]
```diff
@@ -1,5 +1,12 @@
 #include "script_component.hpp"
 
+call FUNC(getMapData);
+
+// Randomization
+GVAR(temperatureShift) = 3 - random 6;
```
Comment (esteldunedain) [inline addons/weather/XEH_postInit.sqf]: For clarity these parameters should only be inited on the server

File: addons/weather/XEH_postInit.sqf [hunk 2]
```diff
@@ -13,23 +20,7 @@ GVAR(rain_current_range) = -1+(random 2);
...
-GVAR(wind_current_range_dir) = -1+(random 2);
-GVAR(wind_next_period) = -1; //ceil((2+random(5))/(GVAR(overcast_multiplier)/10));
-GVAR(wind_next_major_period) = -1;
-GVAR(wind_period_count) = 0;
-GVAR(wind_major_period_count) = 0;
-GVAR(wind_total_time) = 0;
-GVAR(wind_period_start_time) = time;
-
-call FUNC(getMapData);
+call FUNC(initWind);
```
Comment (esteldunedain) [inline addons/weather/XEH_postInit.sqf]: This function too, only on server

**Comment by sunrrrise** [comment]
Do you plan to make it optional and let the mission maker decide which "source" of weather is primary (weather from editor/ACE or weather from Zeus/MCC)?

**Comment by esteldunedain** [comment]
> Do you plan to make it optional and let the mission maker decide which "source" of weather is primary 

No, there's no plans for that yet (it was the same on ACE2). Some aspects of vanilla weather is not synced properly across the network, which lead to inconsistent ballistics, etc. Maybe we could at some point allow editor moduls, API calls or Zeus Modules to modify our simulation

**Comment by ulteq** [comment]
> Do you plan to make it optional and let the mission maker decide which "source" of weather is primary?
I suggest you open a issue for this.

**Pull request merged**